### DISCUSSION TITLE
大佬，编制时报错，这种应该怎么解决？

### DISCUSSION THREAD

**Discussion by NPC2000** [LSPosed/LSPosed#2444] [Q&A] [answered]
`FAILURE: Build completed with 27 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':daemon:mergeReleaseArtProfile'.
> Could not resolve all files for configuration ':daemon:releaseRuntimeClasspath'.
   > Could not find io.github.libxposed:service-interface:100.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - https://repo.maven.apache.org/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - file:/home/npc/.m2/repository/io/github/libxposed/service-interface/100/service-interface-100.pom
     Required by:
         project :daemon

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':daemon:checkReleaseDuplicateClasses'.
> Could not resolve all files for configuration ':daemon:releaseRuntimeClasspath'.
   > Could not find io.github.libxposed:service-interface:100.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - https://repo.maven.apache.org/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - file:/home/npc/.m2/repository/io/github/libxposed/service-interface/100/service-interface-100.pom
     Required by:
         project :daemon

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

3: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':daemon:processReleaseMainManifest'.
> Could not resolve all files for configuration ':daemon:releaseRuntimeClasspath'.
   > Could not find io.github.libxposed:service-interface:100.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - https://repo.maven.apache.org/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - file:/home/npc/.m2/repository/io/github/libxposed/service-interface/100/service-interface-100.pom
     Required by:
         project :daemon

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

4: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':daemon:checkReleaseAarMetadata'.
> Could not resolve all files for configuration ':daemon:releaseRuntimeClasspath'.
   > Could not find io.github.libxposed:service-interface:100.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - https://repo.maven.apache.org/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - file:/home/npc/.m2/repository/io/github/libxposed/service-interface/100/service-interface-100.pom
     Required by:
         project :daemon

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

5: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':daemon:mapReleaseSourceSetPaths'.
> Could not resolve all files for configuration ':daemon:releaseRuntimeClasspath'.
   > Could not find io.github.libxposed:service-interface:100.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - https://repo.maven.apache.org/maven2/io/github/libxposed/service-interface/100/service-interface-100.pom
       - file:/home/npc/.m2/repository/io/github/libxposed/service-interface/100/service-interface-100.pom
     Required by:
         project :daemon

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================`

**Comment by yujincheng08** [comment, marked as answer]
参考 github action 的 workflow

**Reply by NPC2000** [reply]
大佬，不好意思，打扰一下。我是使用android studio来编译的，使用的jdk17,gardle8.0.2。我是先git init 之后git clone 下载，并用git submodule update --init --recursive 下载子模块，使用git pull和git status 检查。然后才使用android studio 去打开项目，等同步完之后，运行zipZygiskRelease，就会出现下面这种报错，点击报错里面的链接，跳到的位置就是谷歌404。file:///home/npc/%E5%9B%BE%E7%89%87/%E6%88%AA%E5%9B%BE/error.png

**Reply by NPC2000** [reply]
![error](https://user-images.githubusercontent.com/113622959/226294020-3dce8988-6b84-4f73-966a-0aa81d5d746b.png)

**Reply by yujincheng08** [reply]
https://github.com/LSPosed/LSPosed/blob/55a488b72a0943a29e641dd15515a9a6e99f5043/.github/workflows/core.yml#L77-L83

**Reply by NPC2000** [reply]
![1111](https://user-images.githubusercontent.com/113622959/226294303-3de5633f-8be6-4696-8e69-1d9c7ec831f5.png)

**Reply by NPC2000** [reply]
> https://github.com/LSPosed/LSPosed/blob/55a488b72a0943a29e641dd15515a9a6e99f5043/.github/workflows/core.yml#L77-L83

感谢大佬，我去试试

**Reply by NPC2000** [reply]
> https://github.com/LSPosed/LSPosed/blob/55a488b72a0943a29e641dd15515a9a6e99f5043/.github/workflows/core.yml#L77-L83

大佬，打扰一下，我获取最新版lsposed，已经解决了api和server的问题，也可以成功编译构建lsposed。但是在android studio进行修改的时候没有语法提示。因为项目同步时出现了这个错误，想问一下这个是怎么解决的。
![截图 2023-03-21 16-25-32](https://user-images.githubusercontent.com/113622959/226552382-ff942dea-559e-4f4e-944f-48270e524eee.png)

**Reply by torrentshare** [reply]
大佬，https://dl.google.com/dl/android/maven2/io/github/libxposed/interface/100/interface-100.pom
       - https://jcenter.bintray.com/io/github/libxposed/interface/100/interface-100.pom
这个报错怎么解决的呀？

**Comment by NPC2000** [comment]
> 大佬，https://dl.google.com/dl/android/maven2/io/github/libxposed/interface/100/interface-100.pom - https://jcenter.bintray.com/io/github/libxposed/interface/100/interface-100.pom 这个报错怎么解决的呀？

git clone --depth 1 https://github.com/libxposed/api.git libxposed/api
git clone --depth 1 https://github.com/libxposed/service.git libxposed/service

cd libxposed/api
./gradlew :api:publishApiPublicationToMavenLocal
cd ../service
./gradlew :interface:publishInterfacePublicationToMavenLocal
